src/include/sof/math/trig.h

```diff
@@ -108,17 +127,17 @@ static inline int32_t cos_fixed_32b(int32_t th_rad_fxp)
  */
 static inline int16_t sin_fixed_16b(int32_t th_rad_fxp)
 {
+	int32_t th_cdc_fxp;
 	int32_t sign;
 	int32_t b_yn;
 	int32_t xn;
-	int32_t th_cdc_fxp;
-	cordic_cfg type = EN_16B_CORDIC_SINE;
-	/* compute coeff from angles*/
-	cordic_sin_cos(th_rad_fxp, type, &sign, &b_yn, &xn, &th_cdc_fxp);
+
+	cordic_approx(th_rad_fxp, CORDIC_15B_TABLE_SIZE, &sign, &b_yn, &xn, &th_cdc_fxp);
+
 	th_cdc_fxp = sign * b_yn;
 	/*convert Q1.31 to Q1.15 format*/
 	return sat_int16(Q_SHIFT_RND((sat_int32(Q_SHIFT_LEFT((int64_t)th_cdc_fxp, 30, 31))),
```

Comment:
You should convert directly from Q2.30 to Q1.15. Something like

`return sat_int16(Q_SHIFT_RND(th_cdc_fxp, 30, 15));`

Do the same for all 16 bit functions. It will speed up them a lot as plain C.